### PR TITLE
Introduced `locate_segment_position()` function

### PR DESCRIPTION
**Pull request by arsenius7**
Given that SONATA format stores synapse positions as section ID + relative distance along the section, the task of locating the corresponding segment might be something needed repeatedly.

Not sure if NeuroM is the appropriate place for it, though.

### CODE REVIEW

File: neurom/morphmath.py [hunk 1]
```diff
@@ -81,6 +81,34 @@ def f(a, b, c):
...
+        fraction: path length fraction (0.0 <= fraction <= 1.0)
+        relative_offset: return absolute or relative segment distance
+
+    Returns:
+        (segment ID, segment offset) pair.
+    '''
+    if not (0. <= fraction <= 1.0):
+        raise ValueError("Invalid fraction: %.3f" % fraction)
+    pts = np.array(points)[:, COLS.XYZ]
+    lengths = np.linalg.norm(np.diff(pts, axis=0), axis=1)
```
Comment (arsenius7) [inline neurom/morphmath.py]: I guess I'm supposed to use `point_dist` for calculating segment lengths, but it feels rather clumsy when I need all the lengths at once; hopefully reusing COLS.XYZ range alleviates this deviation.
In fact, I would argue that `point_dist` itself should be reusing NumPy `diff + norm`, but that's not relevant to this change.

File: neurom/fst/sectionfunc.py [hunk 2]
```diff
@@ -123,3 +123,10 @@ def strahler_order(section):
             return max_so_children + 1
         return max_so_children
     return 1
+
+
+def locate_segment_position(section, fraction):
```
Comment (mgeplf) [inline neurom/fst/sectionfunc.py]: section id isn't something we have tried to make constant/public, and doesn't necessarily match what other software expects the section id to be.  I'm hesitant to have this exposed, unless it's understood that the segment id is only stable during one instance of a morphology, and that guarantees aren't made between other invocations/versions/etc

does that make sense?

Comment (arsenius7) [reply neurom/fst/sectionfunc.py]: Whether we want it or not, section IDs are _already_ used for accessing geometry through NeuroM.
I understand that it is problematic, but saying "we give no guarantees" is hardly an option.

Comment (lidakanari) [reply neurom/fst/sectionfunc.py]: That is quite useful for the morphology analysis as well. Thanks for the implementation!

Comment (arsenius7) [reply neurom/fst/sectionfunc.py]: Glad it found reuse so soon :); but please note that Mike is right: section / segment ID semantics is not well defined yet.

Comment (lidakanari) [reply neurom/fst/sectionfunc.py]: Yes, I am aware of that. The need for this was primarily internal so I assume it is consistent for the same morphology internally in NeuroM which works for me at the moment :)

**Pull request merged**